python/pyiceberg/expressions/literals.py

```diff
@@ -110,11 +106,9 @@ def __ge__(self, other) -> bool:
         return self.value >= other.value
 
 
-@singledispatch
```

Comment:
We had to drop `singledispatch` since it doesn't work well with `TypeVars`: https://github.com/python/mypy/issues/13040